### PR TITLE
Migrate HttpServerTest to Armeria

### PR DESCRIPTION
**Pull request by anuraaga**
I randomly noticed the okhttp instrumentation tests never run against the baseline version because it picks up the higher version from testing-common. Like we did for the server, we should shade the client too to prevent testing components from interacting with instrumentation. Since Armeria's already shading, switching the client is simplest I guess.

I'll follow up to migrate remaining usages.

### CODE REVIEW

File: instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ServerTest.groovy [hunk 1]
```diff
@@ -71,15 +71,17 @@ class Netty41ServerTest extends HttpServerTest<EventLoopGroup> implements AgentT
                       response = new DefaultFullHttpResponse(HTTP_1_1, HttpResponseStatus.valueOf(endpoint.status), content)
                       break
                     case INDEXED_CHILD:
+                      content = Unpooled.EMPTY_BUFFER
```
Comment (anuraaga) [inline instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ServerTest.groovy:74]: @trustin @minwoox @ikhoon OkHttp client worked fine with the old code but Armeria seems to require the content length to be set. Is it expected?

**Review by anuraaga — COMMENTED** [review]
@trustin @minwoox @ikhoon If you have time to skim this to make sure I didn't miss any more recent Armeria idioms that would be nice. Thanks!

File: instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ServerTest.groovy (hunk 1 above)
Comment (anuraaga) [reply instrumentation/netty/netty-4.1/javaagent/src/test/groovy/Netty41ServerTest.groovy:74]: Ah to clarify the behavior is that Armeria client times out (timeout is set to 1 minute). Just doesn't seem to treat the response as complete

File: testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpServerTest.groovy [hunk 2]
```diff
@@ -430,22 +394,21 @@ abstract class HttpServerTest<SERVER> extends InstrumentationSpecification imple
...
     when:
     count.times { index ->
       def job = {
         latch.await()
-        def url = HttpUrl.get(endpoint.resolvePath(address)).newBuilder()
-          .query("${ServerEndpoint.ID_PARAMETER_NAME}=$index")
-          .build()
-        Request.Builder builder = request(url, "GET", null)
+        HttpRequestBuilder request = HttpRequest.builder()
+          .get(endpoint.resolvePath(address).toString().replace("http://", "h1c://"))
```
Comment (trask) [inline testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpServerTest.groovy:406]:
```suggestion
          // Force HTTP/1 via h1c so upgrade requests don't show up as traces
          .get(endpoint.resolvePath(address).toString().replace("http://", "h1c://"))
```

**Review by trask — APPROVED** [review]

**Comment by minwoox** [comment]
Looks perfect! 👍

**Pull request merged**